app/services/onlinetesting/Phase2TestService.scala

```diff
@@ -95,6 +95,25 @@ trait Phase2TestService extends OnlineTestService with Phase2TestConcern with Sc
     }
   }
 
+  def verifyAccessCode(email: String, accessCode: String): Future[Option[String]] = {
+    (for {
+      userId <- cdRepository.findUserIdByEmail(email)
+      testGroupOpt <- phase2TestRepo.getTestGroupByUserId(userId)
+    } yield {
+      testGroupOpt.flatMap { testGroup =>
+        val eTrayTest = testGroup.activeTests.head
```

Comment:
I know that technically this will always be a list of 1 item for etray - but what if it's empty 😱